internal/extproc/translator/openai_gcpvertexai.go

```diff
@@ -271,9 +271,34 @@ func (o *openAIToGCPVertexAITranslatorV1ChatCompletion) openAIMessageToGeminiMes
 		SystemInstruction: systemInstruction,
 	}
 
+	// Apply vendor-specific fields after standard OpenAI-to-Gemini translation.
+	// Vendor fields take precedence over translated fields when conflicts occur.
+	o.applyVendorSpecificFields(openAIReq, &gcr)
+
 	return gcr, nil
```

Comment:
Can you make this `openAIMessageToGeminiMessage` function return pointer? 

Now that we call applyVendorSpecificFields, we escape `gcp.GenerateContentRequest` to heap here inside this function (depending on how Go compiler works though. Escape analysis might work this case but we cannot be sure.) Since at the end of the day, at the call site of openAIMessageToGeminiMessage, we also end up escaping `gcp.GenerateContentRequest` when calling json.Marshal, returning already a pointer from here makes sense because that allows us to have only one escape vs twice